ssl/t1_lib.c

```diff
@@ -1049,7 +1049,8 @@ static int gid_cb(const char *elem, int len, void *arg)
         return 0;
     if (garg->gidcnt == garg->gidmax) {
         uint16_t *tmp =
-            OPENSSL_realloc(garg->gid_arr, garg->gidmax + GROUPLIST_INCREMENT);
+            OPENSSL_realloc(garg->gid_arr, garg->gidmax +
```

Comment:
Did you intend for the operator precedence to do this?
40 + (40 * 2) 
Which is not the same as  (40 + 40) * 2
So you are adding 0.5 * GROUPLIST_INCREMENT currently..

Reply:
As gidmax gets large the halfing will cause it to get smaller I think :)

Reply:
> Did you intend for the operator precedence to do this?

Nope. Thanks for the catch.

> So you are adding 0.5 * GROUPLIST_INCREMENT currently..
> As gidmax gets large the halfing will cause it to get smaller I think :)

Well it was "40 + 40" before the patch... Seems even less to me: Following your logic, maybe even exponential reduction in size as group lists get larger? :)